### ISSUE TITLE
Fatal game crash while using the logic programmer, when clicking "show all recipes" from JEI GUI

### ISSUE THREAD

**Issue by Jack-McKalling** [CyclopsMC/IntegratedDynamics#861]
#### Issue type:

- :bug: Bug

____

#### Short description:

When trying to insert an item from JEI into a variable card with the Logic Programmer, if you click on the "Show all recipes" link at the top of the JEI GUI for any particular recipe, it will instantly crash the whole game.

#### Steps to reproduce the problem:

1. Create a new world
2. Open JEI and search for `@integr` so you see all items from this mod
3. CTRL+click the JEI config button on the right of the search field to enter cheat mode
4. Right-click the portable logic programmer on the right and put it on the hotbar
5. Disable cheat mode again
6. Open the logic programmer, and select:
   - List operator
   - Browse to "Items" type
   - Click "+" to add element
7. Click on an item in JEI to see its crafting recipe, e.g. the variable card
8. Click the "Show all recipes" link
    (instead of proceding to click the "+" for inserting the variable card as an "item" in the list)
   --- crash ---

#### Expected behaviour:

The Logic Programmer should let JEI process the link for browsing to a different recipe collection, instead of assuming it is a transfer handler for an item.

____

#### Versions:

- This mod: 1.0.8
- Minecraft: 1.12.2
- Forge: Sky Factory 4.1.0 (I don't know where to find forge version)

#### Log file:

https://gist.github.com/Jack-McKalling/09a02de0dbb8f0b0d9d2635066219337

**Comment by rubensworks** [comment]
Please try again with the latest ID version, you're running quite a few versions behind.

**Comment by Jack-McKalling** [comment]
I tried with the latest version 1.1.6, in ATLauncher's Vanilla Forge pack, with just ID and JEI installed, with the suggested "Common Capabilities" and "Cyclops Core".

It still happens exactly like I described, it's not even related to Sky Factory that I was initially experiencing this in. (which I did assume so from the error in the crash log).

Versions:
CommonCapabilities-1.12.2-2.4.6
CyclopsCore-1.12.2-1.6.4
IntegratedDynamics-1.12.2-1.1.6
jei_1.12.2-4.15.0.289
